rules/opentitan_test.bzl

```diff
@@ -452,3 +454,140 @@ def opentitan_functest(
             # For more see https://bazel.build/reference/be/general#test_suite.tags
         ],
     )
+
+def _opentitan_sram_fpga_cw310_test(ctx):
+    test_script = '''#!/usr/bin/env bash
+set -e
+
+COLOR_RED='\\x1b[0;31m'
+COLOR_GREEN='\\x1b[0;32m'
+COLOR_PURPLE='\\x1b[0;35m'
+
+function prefix_lines() {{
+    LABEL="$1"
+    COLOR="$2"
+    COLOR_RESET='\\x1b[m'
+    sed -Eu "s/(.*)/$COLOR[$LABEL]$COLOR_RESET \\1/"
+}}
+
+./opentitantool load-bitstream --rom-kind=TestRom test_rom.bit
+
+(openocd -f /usr/share/openocd/scripts/interface/ftdi/olimex-arm-usb-tiny-h.cfg \\
+    -c "adapter speed 500; transport select jtag; reset_config trst_and_srst" \\
+    -f {openocd_earlgrey_config} 2>&1 \\
+    | prefix_lines OPENOCD "$COLOR_PURPLE") &
+
+(/tools/riscv/bin/riscv32-unknown-elf-gdb --command=gdb_script 2>&1 \\
+    | prefix_lines GDB "$COLOR_GREEN") &
+
+./opentitantool console --exit-success='{exit_success_pattern}' \\
```

Comment:
I don't have a good `--exit-failure` pattern, so I didn't specify one.

I guess if SRAM execution fails, this test will eventually time out and possibly block up CI.

I'll pass a `--timeout` parameter to `opentitantool console`.